docs/src/appendix/appendix.md

```diff
@@ -9,6 +9,7 @@ section: "chisel3"
 This section covers some less-common Chisel topics.
 
 * [Differences between Chisel3 and Chisel2](chisel3-vs-chisel2)
+* [Notes on Chisel Plugin](chisel-plugin)
```

Comment:
this will cause this to show up on website and will need a corresponding chisel-lang website PR

Reply:
Removed the link.
Moved the chisel3-plugin.md to a README.md in the plugin folder
Noted that the information there is for developers working on chisel3 internals